### PR TITLE
TASK: Implement PSR-7 interface methods in Flow's HTTP

### PR DESCRIPTION
**Pull request by kitsunet**

### CODE REVIEW

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/Collection.php [hunk 1]
```diff
@@ -165,7 +165,7 @@ public function getObjects()
 
         // TODO: Implement filter manipulation here:
         // foreach ($objects as $object) {
-        // 	$object->setStream(function() { return fopen('/tmp/test.txt', 'rb');});
+        // 	$object->openStream(function() { return fopen('/tmp/test.txt', 'rb');});
```
Comment (dfeyer) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/Collection.php]: Is this really related to this PR ?

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/Storage/Object.php [hunk 2]
```diff
@@ -217,7 +217,7 @@ public function setMd5($md5)
     public function setStream($stream)
     {
         if (!is_resource($stream) && !$stream instanceof \Closure) {
-            throw new \InvalidArgumentException(sprintf('setStream() expects a stream or Closure, %s given.', gettype($stream)), 1416311979);
+            throw new \InvalidArgumentException(sprintf('openStream() expects a stream or Closure, %s given.', gettype($stream)), 1416311979);
```
Comment (dfeyer) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/Storage/Object.php]: Is this really related to this PR ?

**Comment by dfeyer** [comment]
As always your are fast as starting new PR ;)

I think it's a good start, we have a project where we use the current request implementation heavily, I will try to test this PR as soon as possible.

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/Storage/Object.php (hunk 2 above)
Comment (kitsunet) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/Storage/Object.php]: nope, no idea what happened there... will rebase

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/Collection.php (hunk 1 above)
Comment (kitsunet) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/Collection.php]: Also no idea, will rebase.

**Comment by kitsunet** [comment]
Nice! Lets continue improving on this. We also need some stream implementation(s) still. I wasn't sure yet if we want to implement them ourselfs our use some existing implementation as a start. guzzle has some nice stream implementations AFAIK.

**Comment by kdambekalns** [comment]
Needs a dependency to PSR-7 interfaces…

**Comment by kitsunet** [comment]
So, rebasing to master is a real pain, tried to do it yesterday evening, will try again today. Probably need to recreate the changes by hand and push freshly @dfeyer @mficzel very probably that will destroy history and leave a single commit, I hope that's ok.

**Comment by mficzel** [comment]
Offcourse that is ok

**Comment by kitsunet** [comment]
Replaced by #761
